src/js/themes/media-theme-template.js

```diff
@@ -0,0 +1,117 @@
+import { defineCustomElement } from '../utils/defineCustomElement.js';
+import MediaTheme from './media-theme.js';
+import { InnerTemplatePart, TemplateInstance, AttrPart } from './template-parts.js';
+
+class MediaThemeTemplate extends MediaTheme {
+  constructor() {
+    super();
+  }
+
+  connectedCallback() {
+    this.template = this.getRootNode().querySelector(`#${this.getAttribute('template')}`);
+
+    this.templateInstance = new TemplateInstance(
+      this.template,
+      this.props,
+      mediaThemeProcessor
+    );
+
+    this.shadowRoot.append(this.templateInstance);
+
+    const observer = new MutationObserver(() => this.render());
+    observer.observe(this, { attributes: true });
+  }
+
+  get props() {
+    const props = {};
+    for (let attr of this.attributes) {
+      if (attr.value != null) {
```

Comment:
issue(blocking): This assumes that values are only ever a string or a boolean, and all based on the value, which also means e.g. `''` **_always_** means `true`, even in cases like `title=''`

Reply:
yes, this is an issue that is solved with casting in the template language.
we talked a bit about yesterday, it looks like it will be a requirement to solve this use case cc @heff 

syntax is up for discussion. I added something like `{{ title | string }}` similar to Nunjucks I believe